apps/postgres-new/components/button-copy-code.tsx

```diff
@@ -0,0 +1,31 @@
+'use client'
+import { useState } from "react";
+import { Button } from "./ui/button";
+import { Check, Copy } from "lucide-react";
+
+interface ButtonCopyCodeProps {
+    code?: string;
+}
```

Comment:
A few things:

1. Let's make this a type (vs interface) just to be consistent with the other components
2. Any good reason to make `code` optional? Probably a clearer contract to make this required (and then upstream components can conditionally render the copy button if the value is optional.
3. Maybe we just call this `CopyButton` with a `value` prop so that it is more generic and can be used outside code block if we want in the future